Adding LocationComponent camera mode example

langsmith opened the pull request:
This pr resolves https://github.com/mapbox/mapbox-android-demo/issues/969 by adding an example that show the `LocationComponent`'s various camera mode options.  The example is a close copy of [the Maps SDK test app's `LocationComponent` camera options example](https://github.com/mapbox/mapbox-gl-native/blob/master/platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/location/LocationModesActivity.java): 

Related: https://github.com/mapbox/android-docs/issues/854

![ezgif com-resize](https://user-images.githubusercontent.com/4394910/59537829-8d70a680-8eac-11e9-8b36-10a5872a8e13.gif)

langsmith commented:
@riastrad , I've opened this pr to get things started on https://github.com/mapbox/mapbox-android-demo/issues/969. Not sure whether you still want to or have the bandwidth to run with this. No worries if not.

riastrad commented:
@langsmith realizing that I'm likely not going to get to this anytime soon. If you have the time and want to run with it, please feel free!

langsmith commented:
Ok, thanks for letting me know, @riastrad

langsmith commented:
Ok @LukasPaczos , this ready for review.

langsmith commented:
Ok @LukasPaczos ,  I just pushed a commit that removes the options menu stuff (style toggling, fps throttling, etc.), added code comments, and also added 2 missing modes (none_compass & none_gps).

langsmith commented:
Final version of the example looks like 👇 

![ezgif com-resize (1)](https://user-images.githubusercontent.com/4394910/59938959-4b46e800-940b-11e9-8058-cd6d2a34c4b0.gif)
